cmd/api/src/api/middleware/middleware.go

```diff
@@ -147,19 +152,16 @@ func ContextMiddleware(next http.Handler) http.Handler {
 	})
 }
 
-func parseUserIP(r *http.Request) string {
-	IPAddress := r.Header.Get("X-Real-Ip")
-	if IPAddress == "" {
-		IPAddress = r.Header.Get("X-Forwarded-For")
-	}
-	if IPAddress == "" {
-		if parsedUrl, err := url.Parse(r.RemoteAddr); err != nil {
-			log.Errorf("error parsing IP address from RemoteAddr: %s", err)
-		} else {
-			IPAddress = parsedUrl.Hostname()
-		}
+func parseUserIP(r *http.Request) (string, error) {
+	if ipAddress := r.Header.Get("X-Forwarded-For"); ipAddress != "" {
+		return strings.Split(ipAddress, ",")[0], nil
+	} else if parsedUrl, err := url.Parse(r.RemoteAddr); err != nil {
+		return "", fmt.Errorf("error parsing IP address from RemoteAddr: %s", err)
+	} else if hostName := parsedUrl.Hostname(); hostName == "" {
+		return "", fmt.Errorf("hostname not found in URL: %s", parsedUrl.String())
+	} else {
+		return parsedUrl.Hostname(), nil
```

Comment:
We already have the `hostName` var, but otherwise this logic seems good. I'd like to find out for sure if we need to attach any additional information on how the IP was derived for audit purposes, but that can be cleaned up in a follow up